Trace analytics ingestion End-to-end tests have been flaky

**Describe the bug**
Recently I am seeing more frequent CI failure on e2e tests due to UnprocessedRequestException when Armeria client sending the requests.
```
Task :data-prepper-core:rawSpanEndToEndTest

com.amazon.dataprepper.integration.EndToEndRawSpanTest > testPipelineEndToEnd FAILED
    io.grpc.StatusRuntimeException at EndToEndRawSpanTest.java:156
        Caused by: com.linecorp.armeria.client.UnprocessedRequestException at UnprocessedRequestException.java:45
            Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException
                Caused by: java.net.ConnectException at Errors.java:155
```
**To Reproduce**
This is produced by github CI.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment (please complete the following information):**
 - OS: linux
 - Version: github

**Additional context**
Add any other context about the problem here.

--- a/data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java
+++ b/data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndRawSpanTest.java
@@ -18,6 +18,8 @@ import com.fasterxml.jackson.core.type.TypeReference;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import com.google.protobuf.ByteString;
 import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.retry.RetryRule;
+import com.linecorp.armeria.client.retry.RetryingClient;
 import com.linecorp.armeria.internal.shaded.bouncycastle.util.encoders.Hex;
 import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
 import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpc;
@@ -152,8 +154,10 @@ public class EndToEndRawSpanTest {
     }
 
     private void sendExportTraceServiceRequestToSource(final int port, final ExportTraceServiceRequest request) {
-        Clients.newClient(String.format("gproto+http://127.0.0.1:%d/", port),
-                TraceServiceGrpc.TraceServiceBlockingStub.class).export(request);
+        TraceServiceGrpc.TraceServiceBlockingStub client = Clients.builder(String.format("gproto+http://127.0.0.1:%d/", port))
+                .decorator(RetryingClient.newDecorator(RetryRule.failsafe()))
+                .build(TraceServiceGrpc.TraceServiceBlockingStub.class);
+        client.export(request);
     }
 
     private List<Map<String, Object>> getSourcesFromSearchHits(final SearchHits searchHits) {

--- a/data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java
+++ b/data-prepper-core/src/integrationTest/java/com/amazon/dataprepper/integration/EndToEndServiceMapTest.java
@@ -27,6 +27,8 @@ import java.util.stream.Collectors;
 import java.util.stream.Stream;
 
 import com.linecorp.armeria.client.Clients;
+import com.linecorp.armeria.client.retry.RetryRule;
+import com.linecorp.armeria.client.retry.RetryingClient;
 import io.opentelemetry.proto.collector.trace.v1.ExportTraceServiceRequest;
 import io.opentelemetry.proto.collector.trace.v1.TraceServiceGrpc;
 import io.opentelemetry.proto.common.v1.AnyValue;
@@ -132,8 +134,10 @@ public class EndToEndServiceMapTest {
     }
 
     private void sendExportTraceServiceRequestToSource(final int port, final ExportTraceServiceRequest request) {
-        Clients.newClient(String.format("gproto+http://127.0.0.1:%d/", port),
-                TraceServiceGrpc.TraceServiceBlockingStub.class).export(request);
+        TraceServiceGrpc.TraceServiceBlockingStub client = Clients.builder(String.format("gproto+http://127.0.0.1:%d/", port))
+                .decorator(RetryingClient.newDecorator(RetryRule.failsafe()))
+                .build(TraceServiceGrpc.TraceServiceBlockingStub.class);
+        client.export(request);
     }
 
     private List<Map<String, Object>> getSourcesFromIndex(final RestHighLevelClient restHighLevelClient, final String index) throws IOException {

